src/server/components/dashboard/helpers.ts

```diff
@@ -57,3 +61,27 @@ export async function getInitiateFormRunnerSessionToken(
   logger.info(`token: ${token}`);
   return token;
 }
+
+export const pageTitles: { [key: string]: string } = {
+  [dashboardRoutes.usersEdit]: "list management - edit user",
+  [dashboardRoutes.usersList]: "list management - user list",
+  [dashboardRoutes.lists]: "list management - all provider lists",
+  [dashboardRoutes.listsEdit]: "list management - :serviceType in :country - edit provider list",
+  [dashboardRoutes.listsItems]: "list management - :serviceType in :country - provider list",
+  [dashboardRoutes.listsItem]: "list management - :serviceType in :country - provider details",
+  [dashboardRoutes.listsItemDelete]: "list management - :serviceType in :country - confirm delete provider",
+  [dashboardRoutes.listsItemPublish]: "list management - :serviceType in :country - confirm publish list item",
+  [dashboardRoutes.listsItemRequestChanges]: "list management - :serviceType in :country - confirm request changes to provider",
+  [dashboardRoutes.listsItemUpdate]: "list management - :serviceType in :country - confirm update provider",
+  [sitemapRoute]: "site map",
+  [authRoutes.login]: "list management - login",
+  [authRoutes.logout]: "list management - logout",
+}
+
+export function getPageTitle(pageTitle: string, serviceType?: string, country?: string): string {
+  let result = pageTitle;
+  if (serviceType) {
+    result = result.replace(":serviceType", serviceName(serviceType));
+  }
+  return result.replace(":country", country ?? "Undefined");
+}
```

Comment:
I'm quite curious as to why both the `serviceType` and `country` arguments are optional.

I ask because if either of them do not exist, the string that is produced for the title is somewhat malformed.

If that's the case, in the analytics dashboard there could be visits to pages that have the word `Undefined` in them.

Is this the intention for the return value of the `getPageTitle()` function?

Reply:
In the event where there are field errors, the country from the body is populated back in the list JSON object.  If the issue originates from an empty / invalid country field then the user is shown an error message and the page title would include Undefined.  The idea is that GA would then show these pages and they could be used to identify the frequency of any such invalid instances.

Reply:
Looks like `getPageTitle` has been removed so this is less of an issue 😁